com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelController.java

```diff
@@ -27,55 +33,99 @@
 import com.eclipsesource.modelserver.emf.configuration.ServerConfiguration;
 import com.google.inject.Inject;
 
-import io.javalin.Context;
 import io.javalin.apibuilder.CrudHandler;
-import io.javalin.json.JavalinJackson;
+import io.javalin.http.BadRequestResponse;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.json.JavalinJackson;
 
 public class ModelController implements CrudHandler {
 
+	private static final Logger LOG = Logger.getLogger(ModelController.class);
+
 	@Inject
 	private ResourceManager resourceManager;
 	@Inject
 	private ServerConfiguration serverConfiguration;
 
-	private ResourceSet resourceSet= new ResourceSetImpl();
+	private ResourceSet resourceSet = new ResourceSetImpl();
+	private Map<URI, EObject> modelRepository = new HashMap<URI, EObject>();
 	
 	public ModelController() {
 		JavalinJackson.configure(EMFJsonConverter.setupDefaultMapper());
 	}
 
 	@Override
 	public void create(Context ctx) {
-		// TODO Auto-generated method stub
+		try {
+			EObject model = ctx.bodyAsClass(EObject.class);
+			
+			EStructuralFeature name = model.eClass().getEStructuralFeature("name");
+			String modelName = model.eGet(name).toString();
+			
+			this.modelRepository.put(getUriFromFilepath(modelName.replaceAll(" ", "")), model);
+		} catch (BadRequestResponse r) {
+			handleError(ctx, 400, r.getMessage());
+		} catch (NullPointerException e) {
+			handleError(ctx, 400, "Create new model failed: Model identifier (name) is missing");
+		}
 	}
 
 	@Override
 	public void delete(Context ctx, String modeluri) {
-		// TODO Auto-generated method stub
+		this.modelRepository.remove(getUriFromFilepath(modeluri));
 	}
 
 	@Override
 	public void getAll(Context ctx) {
-		// TODO Auto-generated method stub
+		ctx.json(this.modelRepository.entrySet());
 	}
 
 	@Override
 	public void getOne(Context ctx, String modeluri) {
-		loadModel(modeluri).map(ctx::json).orElse(ctx.status(404));
+		EObject model = loadModel(modeluri);
+		if (model != null)
+			ctx.json(model);
+		else {
+			handleError(ctx, 404, "Model not found");
+		}
 	}
 
 	@Override
 	public void update(Context ctx, String modeluri) {
-		// TODO Auto-generated method stub
+		EObject model = ctx.bodyAsClass(EObject.class);
+		this.modelRepository.put(getUriFromFilepath(modeluri), model);
 	}
+	
+	public Handler getModelUris = ctx -> {
+		Set<String> modelUris = new HashSet<String>();
+		for(URI uri: this.modelRepository.keySet()){
+			modelUris.add(uri.toString());
+		}
+		ctx.json(modelUris);
+	};
 
-	private Optional<EObject> loadModel(String filePath) {
+	private EObject loadModel(String filePath) {
+		final URI modelUri = getUriFromFilepath(filePath);
+		
+		if (!this.modelRepository.containsKey(modelUri)) {
```

Comment:
Use `computeIfAbsent` for updating the map. The final `get` can return null, so we should map accordingly to a 404, e.g. via `Optional.ofNullable(modelRepository.get(modelUri))`